src/main/java/lotto/collections/LottoResult.java

```diff
@@ -0,0 +1,27 @@
+package lotto.collections;
+
+import java.util.Map;
+import java.util.Optional;
+import lotto.util.LottoProfitCalculator;
+
+public class LottoResult {
+
+  private final Map<Integer, Integer> lottoStatistics;
+
+  public LottoResult(Map<Integer, Integer> lottoStatistics) {
+    this.lottoStatistics = lottoStatistics;
+  }
+
+  public double getLottoProfitRate(final int lottoTicketSize) {
+    double totalProfit = RewardType.calculateProfit(lottoStatistics);
+    return new LottoProfitCalculator(totalProfit, lottoTicketSize).getConvertProfit();
+  }
+
+  public int countTicketsByRewardType(RewardType rewardType) {
+    return Optional.ofNullable(lottoStatistics.get(rewardType.getCode())).orElse(0);
+  }
```

Comment:
Optional의 사용은 비싸기 때문에 단순히 값 또는 null을 얻을 목적이라면 Optional 대신 null 비교 하시는걸 권장드려요 :)
```suggestion
  public int countTicketsByRewardType(RewardType rewardType) {
      Integer count = lottoStatistics.get(rewardType.getCode());
    if (count != null) {
      return count;
    }
    return 0;
  }
```

다음 링크를 참고하시면 좋을 것 같아요!
http://homoefficio.github.io/2019/10/03/Java-Optional-%EB%B0%94%EB%A5%B4%EA%B2%8C-%EC%93%B0%EA%B8%B0/

Reply:
> API Note: Optional is primarily intended for use as a method return type where there is a clear need to represent “no result,” and where using null is likely to cause errors. A variable whose type is Optional should never itself be null; it should always point to an Optional instance.

우리 입장에서는 값을 얻는다는 의미도 있지만, count의 결과값이 절대 ```null```이 되면 안되기에 Optional 사용도 적절하지 않을까요? 만약 이런 상황에서 사용하지 마라고 한다면, Optional을 언제 사용해야 하는 지 궁금합니다.

Reply:
진형님께서 말씀해주신대로 Optional에 대해 API도 확인하고 인지하고 쓰신다면 상관없을 것 같아요! Optional에 대해 무분별하게 사용할 경우에 대비하여 드렸던 리뷰였습니다 🙏 

Optional은 값을 포장하고 다시 풀고, 값이 없을 때 대체하는 값을 넣는 등의 오버헤드가 있으므로, 무분별하게, 적절하지 않게 사용된다면 성능 저하가 있을 수 있기 때문에 가능한 지양하는게 좋다고 권장 드리고 있습니다.

또한 Optional은 검사 예외와 취지가 비슷하기 때문에 간혹 구현 실수로 비검사 예외를 던지거나 null을 반환 한다면 사용자가 그 사실을 인지하지 못해 런타임에서 예상치 못한 장애로 발전할 수도 있기 때문입니다 :)

Reply:
감사합니다. 주신 리뷰 덕분에 Optional에 대해 조금 더 찾아보았고 생각보다 많은 안티패턴이 존재한다는 것을 알게 되었네요. 앞으로 Optional 타입 사용에 더 신중해질 것 같습니다.